src/main/java/edu/ie3/datamodel/io/connectors/CsvFileConnector.java

```diff
@@ -51,6 +57,59 @@ public CsvFileConnector(Path baseDirectoryName, FileNamingStrategy fileNamingStr
     this.fileNamingStrategy = fileNamingStrategy;
   }
 
+  @SuppressWarnings("unchecked")
+  public void validate(Map<Class<?>, SourceValidator<?>> pairs, String csvSep) {
+    List<FactoryException> exceptions = new ArrayList<>();
+
```

Comment:
If we want to match the entity class with the validator type class, we should not expect a map as an argument, but just a single class/validator combination like within other connectors.

Secondly, how about making csvSep part of the `CsvFileConnector` class (passed via constructor). This way, all `validate` methods have the same method signature and can be required by `DataSource`.
Or, alternatively to this second point:
I think semantically it'd make more sense to have the `validate` method inside the `DataSource`s (like `CsvDataSource`) instead of the `DataConnector`s (like `CsvFileConnector`). So there would be a `validate` method required by `DataSource`. Would this be possible?

Finally we can then call the validate method from `*Source` rather than `Csv*Source` (for most entities). This will also come in handy when SQL sources are introduced.

Reply:
It is not possible to put the validation method inside the `DataSource`, because we need the information from the `Connectors`.
Also if we have the validation inside the `Connector`s we don't need to implement validation methods for each type of `Source` (like `DataSource`, `IdCoordinateSource`, ....).